### DISCUSSION TITLE
Whisper's generated subtitles don't correspond to the video's playback time point

### DISCUSSION THREAD

**Discussion by ioudove** [openai/whisper#1720] [Q&A] [answered]
The subtitles I generate through Whisper don't accurately correspond to the speech in the video when I play the video and the subtitles are displayed. How can I solve this problem?
My usage: The m4a file I transcribed with Whisper exceeds the official limit of 25M, but Whisper still transcribes my file completely. The only problem now is that the subtitles are not synchronized in time during playback.
I am running Whisper with Colab. For the Colab settings, I am using the [example](https://colab.research.google.com/github/openai/whisper/blob/master/notebooks/LibriSpeech.ipynb) provided by Whisper on Github directly.
Also, I want to provide Whisper with a thesaurus of my own. This will greatly improve the accuracy of Whisper's transcription. How should this be done?
Can anyone help me with this? Thanks.

**Comment by phineas-pta** [comment]
> official limit of 25M

there's no such things

> example provided by Whisper

those notebooks are for showcase of whisper capabilities, they're bloated for simple use case

a very simple notebook only need 2 cells
```
!pip install openai-whisper
```
```
!whisper YOUR_AUDIO.m4a --model large --language zh --task transcribe --output_format srt
```

**Reply by ioudove** [reply]
Thank you very much for your answer, it was very helpful.
I have two more questions:
1, Do you have a solution for the problem of inaccurate correspondence between the subtitle time and the video playback time?
2, What can I do to provide Whisper with my own customized thesaurus so that I don't have to change the words in the Srt file every time.
Thank you.

**Reply by phineas-pta** [reply, marked as answer]
1. hallucinations problem -> must remove silence

2. use `--initial-prompt`

**Reply by ioudove** [reply]
Hi, I'm here to give you feedback. I tested it, and it turns out that my lexicon isn't working. Whisper isn't generating subtitles according to my lexicon. Is this due to my own mistake?

!whisper 1111111.wav --model large --language zh --initial_prompt "5 10 20 30 60 120 10均线 5均线 20均线 30均线 60均线 季线 MACD KDJ 回踩" --output_format srt

**Comment by ioudove** [comment]
> 1. hallucinations problem -> must remove silence
> 2. use `--initial-prompt`

Thanks for the pointers. Since I don't know anything about programming, please ask. Is this how I should add my own thesaurus?
 --initial-prompt one two three......
Thank you very much  ：D

**Reply by phineas-pta** [reply]
`--initial-prompt "......."`

**Reply by ioudove** [reply]
Thank you very very much :D

**Reply by ioudove** [reply]
Hi , I typed it in the format you provided and it came back with this：
whisper: error: unrecognized arguments: --initial-prompt 5,10,20,30,60,120,10均线,5均线,20均线,30均线,60均线,季线
I tried that too, replacing the commas between words with the space bar, and it still prompts this error as well

**Reply by phineas-pta** [reply]
在双引号中

**Reply by ioudove** [reply]
你是中国人么？：D
我加了双引号，依然提示错误。
 --initial-prompt "5,10,20,30,60,120,10均线,5均线,20均线,30均线,60均线,季线"

**Reply by phineas-pta** [reply]
哎呀我的错 🥲 `--initial_prompt`

不是中国人，但懂一点中文

**Reply by ioudove** [reply]
好的，再次感谢您的，真的非常感谢，明天我尝试下 ：D
